packages/material-ui-lab/src/DateRangePickerDay/DateRangePickerDay.tsx

```diff
@@ -43,96 +149,98 @@ const startBorderStyle = {
   borderBottomLeftRadius: '50%',
 };
 
-export type DateRangePickerDayClassKey =
-  | 'root'
-  | 'rangeIntervalDayHighlight'
-  | 'rangeIntervalDayHighlightStart'
-  | 'rangeIntervalDayHighlightEnd'
-  | 'day'
-  | 'dayOutsideRangeInterval'
-  | 'dayInsideRangeInterval'
-  | 'notSelectedDate'
-  | 'rangeIntervalPreview'
-  | 'rangeIntervalDayPreview'
-  | 'rangeIntervalDayPreviewStart'
-  | 'rangeIntervalDayPreviewEnd';
-
-export const styles: MuiStyles<DateRangePickerDayClassKey> = (
-  theme,
-): StyleRules<DateRangePickerDayClassKey> => ({
-  root: {
-    '&:first-child $rangeIntervalDayPreview': {
-      ...startBorderStyle,
-      borderLeftColor: theme.palette.divider,
-    },
-    '&:last-child $rangeIntervalDayPreview': {
-      ...endBorderStyle,
-      borderRightColor: theme.palette.divider,
-    },
+const DateRangePickerDayRoot = styled(
+  'div',
+  {},
+  {
+    name: 'MuiDateRangePickerDay',
+    slot: 'Root',
+    overridesResolver: (props, styles) => styles.root,
   },
-  rangeIntervalDayHighlight: {
-    borderRadius: 0,
-    color: theme.palette.primary.contrastText,
-    backgroundColor: alpha(theme.palette.primary.light, 0.6),
-    '&:first-child': startBorderStyle,
-    '&:last-child': endBorderStyle,
+)<{ styleProps: StyleProps }>(({ theme, styleProps }) => ({
+  [`&:first-of-type .${dateRangePickerDayClasses.rangeIntervalDayPreview}`]: {
+    ...startBorderStyle,
+    borderLeftColor: theme.palette.divider,
   },
-  rangeIntervalDayHighlightStart: {
+  [`&:last-of-type .${dateRangePickerDayClasses.rangeIntervalDayPreview}`]: {
+    ...endBorderStyle,
+    borderRightColor: theme.palette.divider,
+  },
+  ...(styleProps.isHighlighting &&
+    !styleProps.outsideCurrentMonth && {
+      borderRadius: 0,
+      color: theme.palette.primary.contrastText,
+      backgroundColor: alpha(theme.palette.primary.light, 0.6),
+      '&:first-of-type': startBorderStyle,
+      '&:last-of-type': endBorderStyle,
+    }),
+  ...((styleProps.isStartOfHighlighting || styleProps.isStartOfMonth) && {
     ...startBorderStyle,
     paddingLeft: 0,
     marginLeft: DAY_MARGIN / 2,
-  },
-  rangeIntervalDayHighlightEnd: {
+  }),
+  ...((styleProps.isEndOfHighlighting || styleProps.isEndOfMonth) && {
     ...endBorderStyle,
     paddingRight: 0,
     marginRight: DAY_MARGIN / 2,
+  }),
+}));
+
+const DateRangePickerDayRangeIntervalPreview = styled(
+  'div',
+  {},
+  { name: 'MuiDateRangePickerDay', slot: 'RangeIntervalPreview' },
+)<{ styleProps: StyleProps }>(({ theme, styleProps }) => ({
+  // replace default day component margin with transparent border to avoid jumping on preview
+  border: '2px solid transparent',
+  ...(styleProps.isPreviewing &&
+    !styleProps.outsideCurrentMonth && {
+      borderRadius: 0,
+      border: `2px dashed ${theme.palette.divider}`,
+      borderLeftColor: 'transparent',
+      borderRightColor: 'transparent',
+      [`&.${dateRangePickerDayClasses.rangeIntervalDayPreviewStart}`]: {
```

Comment:
Can't we use conditional styles here instead of increasing specificity? Something like:

```
(styleProps.isStartOfPreviewing || styleProps.isStartOfMonth)
```

Reply:
Good point.